src/ziggurat/mapper.clj

```diff
@@ -9,12 +9,12 @@
             [ziggurat.sentry :refer [sentry-reporter]])
   (:import (java.time Instant)))
 
-(defn- send-msg-to-channel [channels message-payload return-code]
+(defn- send-msg-to-channel [channels message-payload return-code headers]
   (when-not (contains? (set channels) return-code)
     (throw (ex-info "Invalid mapper return code" {:code return-code})))
-  (producer/publish-to-channel-instant-queue return-code message-payload))
+  (producer/publish-to-channel-instant-queue return-code message-payload headers))
 
-(defn mapper-func [mapper-fn channels]
+(defn mapper-func [mapper-fn channels headers]
```

Comment:
@roobalimsab Why can't we have headers in message-payload?

Reply:
added